CompatBot/EventHandlers/UsernameValidationMonitor.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Threading.Tasks;
+using CompatBot.Database;
+using CompatBot.Utils;
+using DSharpPlus;
+using DSharpPlus.Entities;
+using DSharpPlus.EventArgs;
+using Microsoft.EntityFrameworkCore;
+
+namespace CompatBot.EventHandlers
+{
+    public static class UsernameValidationMonitor
+    {
+        public static async Task OnUserUpdated(UserUpdateEventArgs args)
+        {
+            await UpdateDisplayName(args.Client, () => args.Client.GetMember(args.UserAfter)).ConfigureAwait(false);
```

Comment:
I'm not sure `OnUserUpdated` is of any interest to us, we're only interested in server nickname change, which will only generate `OnMember` events